pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -361,13 +352,17 @@ public Representation post(Representation entity) {
           // list of FileItems
           items = upload.parseRequest(getRequest());
 
-          for (final Iterator<FileItem> it = items.iterator(); it.hasNext() && !found;) {
-            final FileItem fi = it.next();
-            if (fi.getFieldName() != null) {
-              found = true;
-              dataFile = new File(tempDir, fi.getFieldName());
-              fi.write(dataFile);
+          for (FileItem fileItem : items) {
+            if (!found) {
+              if (fileItem.getFieldName() != null) {
+                found = true;
+                dataFile = new File(tempDir, fileItem.getFieldName());
+                fileItem.write(dataFile);
+              }
+            } else {
+              LOGGER.warn("Got extra file item while pushing segments: " + fileItem.getFieldName());
             }
+            fileItem.delete();
```

Comment:
Also, it will be a good idea to field any exceptions and log a warning so that the upload does not fail if exceptions are thrown for any reason